### PR TITLE
build: migrated python script to typescript for Notification Ops Item

### PR DESCRIPTION
**Pull request by gaudubey**

### CODE REVIEW

File: cdk/package.json [hunk 1]
```diff
@@ -33,15 +31,18 @@
...
+    "@aws-sdk/client-cloudwatch": "^3.495.0",
+    "@aws-sdk/client-eventbridge": "^3.495.0",
+    "@aws-sdk/client-organizations": "^3.495.0",
+    "@aws-sdk/client-sns": "^3.495.0",
+    "@aws-sdk/client-ssm": "^3.495.0",
     "@types/aws-lambda": "^8.10.109",
     "async-retry": "^1.3.3",
-    "aws-cdk-lib": "^2.96.2",
+    "aws-cdk-lib": "^2.115.0",
+    "aws-sdk-client-mock": "^3.0.1",
```
Comment (peermuellerxw) [inline cdk/package.json:42]: aws-sdk-client-mock and aws-sdk-client-mock-jest should stay in "devDepencies"

File: cdk/src/stacks/notifications.ts [hunk 2]
```diff
@@ -18,14 +19,16 @@ export class NotificationsStack extends NestedStack {
...
 
     // NotificationOpsItemCreated
-    const notificationOpsItemCreatedFn = new pythonLambda.PythonFunction(this, 'NotificationOpsItemCreated', {
-      entry: path.join(__dirname, '..', 'functions', 'notification_opsItem_created'),
+    const notificationOpsItemCreatedFn = new NodejsFunction(this, 'NotificationOpsItemCreated', {
+      entry: path.join(__dirname, '..', 'functions', 'notification-opsitem-created.ts'),
       handler: 'handler',
-      runtime: lambda.Runtime.PYTHON_3_9,
+      runtime: lambda.Runtime.NODEJS_18_X,
+      timeout: Duration.minutes(1),
```
Comment (peermuellerxw) [inline cdk/src/stacks/notifications.ts:26]: Maybe reduce the timeout? Before it was just 3 seconds, I believe.

File: cdk/test/functions/notification-opsitem-created.test.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          opsItemId: ops_item_id,
+        },
+        requestParameters: {
+          description: description,
+          title: title,
+        },
+      },
+    }, {});
+
+    expect(snsClientMock).toReceiveCommandTimes(PublishCommand, 1);
```
Comment (peermuellerxw) [inline cdk/test/functions/notification-opsitem-created.test.ts:42]: To improve the test even more you could assert that the contents of the PublishCommand that was received actually contain the body and message.

**Review by peermuellerxw — CHANGES_REQUESTED** [review]

**Review by peermuellerxw — APPROVED** [review]
Looks good. Please rebase/merge from master and I think then this is good to go.

**Review by aperov9 — CHANGES_REQUESTED** [review]

**Pull request merged**